tests/foreman/ui/test_host.py

```diff
@@ -1701,3 +1720,196 @@ def test_positive_delete_libvirt(
+        })
+    gce_cr = entities.AbstractComputeResource().search(
+        query={'search': 'name={}'.format(cr_name)})[0]
+    entities.Image(
+        architecture=default_architecture, compute_resource=gce_cr, name='autogce_img',
+        operatingsystem=module_os, username=vm_user, uuid=gce_template
+    ).create()
+    return gce_cr
+
+
+@pytest.fixture
+def gce_hostgroup(
+        module_org,
+        module_loc,
+        default_partition_table,
+        default_architecture,
+        module_os,
+        module_environment,
+        module_proxy,
+        gce_domain,
+        gce_resource_with_image,
+        module_lce,
+        module_content_view,
+):
+    return entities.HostGroup(
+        architecture=default_architecture,
+        compute_resource=gce_resource_with_image,
+        domain=gce_domain,
+        lifecycle_environment=module_lce,
+        content_view=module_content_view,
+        location=[module_loc],
+        environment=module_environment,
+        puppet_proxy=module_proxy,
+        puppet_ca_proxy=module_proxy,
+        content_source=module_proxy,
+        operatingsystem=module_os,
+        organization=[module_org],
+        ptable=default_partition_table,
+    ).create()
+
+
+@tier4
+@skip_if_not_set('gce')
+def test_positive_gce_provision_end_to_end(
+        session,
+        module_org,
+        module_loc,
+        module_os,
+        gce_domain,
+        gce_hostgroup,
+        gce_client
+):
+    """Provision Host on GCE compute resource
+
+    :id: 8d1877bb-fbc2-4969-a13e-e95e4df4f4cd
+
+    :expectedresults: Host is provisioned successfully
+
+    :CaseLevel: System
+    """
+    name = gen_string('alpha').lower()
+    hostname = u'{0}.{1}'.format(name, gce_domain.name)
+    gceapi_vmname = hostname.replace('.', '-')
+    root_pwd = gen_string('alpha', 15)
+    with Session('gce_tests') as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        # Provision GCE Host
+        try:
+            skip_yum_update_during_provisioning(template='Kickstart default finish')
+            session.host.create({
+                'host.name': name,
+                'host.hostgroup': gce_hostgroup.name,
+                'provider_content.virtual_machine.machine_type': 'g1-small',
+                'provider_content.virtual_machine.external_ip': True,
+                'provider_content.virtual_machine.network': 'default',
+                'operating_system.operating_system': module_os.title,
+                'operating_system.image': 'autogce_img',
+                'operating_system.root_password': root_pwd
+            })
+            wait_for(
+                lambda: entities.Host().search(
+                    query={'search': 'name={}'.format(
+                        hostname)})[0].build_status_label != 'Pending installation',
+                timeout=600,
+                delay=30,
```

Comment:
```suggestion
                delay=10,
```
Just a suggestion to change the `delay` as that would help to save atleast sometime in test.

Reply:
Dont you think its so frequent? May be 20 seconds ?

Reply:
15-20 seconds gap should be fine. please add.